ext/json/ext/generator/generator.c

```diff
@@ -991,6 +991,29 @@ static void generate_json_string(FBuffer *buffer, struct generate_json_data *dat
     fbuffer_append_char(buffer, '"');
 }
 
+static void generate_json_fallback(FBuffer *buffer, struct generate_json_data *data, JSON_Generator_State *state, VALUE obj)
+{
+    VALUE tmp;
+    if (rb_respond_to(obj, i_to_json)) {
+        tmp = rb_funcall(obj, i_to_json, 1, vstate_get(data));
+        Check_Type(tmp, T_STRING);
+        fbuffer_append_str(buffer, tmp);
+    } else {
+        tmp = rb_funcall(obj, i_to_s, 0);
+        Check_Type(tmp, T_STRING);
+        generate_json_string(buffer, data, state, tmp);
+    }
+}
+
+static inline void generate_json_symbol(FBuffer *buffer, struct generate_json_data *data, JSON_Generator_State *state, VALUE obj)
+{
+    if (state->strict) {
```

Comment:
It seems unfortunate to only do this fast path when strict is set, should we maybe always do it?

Reply:
i.e. I doubt anyone relies on `Symbol#to_json` to be called, there are not that many ways to dump a Symbol to JSON

Reply:
Unfortunately we can't, because of this stupid `json/add/symbol`: https://github.com/ruby/json/blob/c472d72fdf1fd8e297cfdaf05b95b1ddcfef63fb/lib/json/add/symbol.rb